libobs/obs-scene.c

```diff
@@ -2082,6 +2084,56 @@ void obs_sceneitem_set_order_position(obs_sceneitem_t *item, int position)
 	obs_scene_release(scene);
 }
 
+void obs_scene_set_items_order(obs_scene_t *scene, int64_t* new_items_order, int items_count)
+{
+	if (!scene || items_count <= 1)
+		return;
+
+	obs_scene_addref(scene);
+	full_lock(scene);
+
+	//create array with items of this scene
+	obs_sceneitem_t **scene_items_cached = bzalloc(items_count * sizeof(obs_sceneitem_t*)); 
+	int index = 0;
+	scene_items_cached[0] = scene->first_item;
+	while (scene_items_cached[index] && index < items_count - 1) {
+		scene_items_cached[index+1] = scene_items_cached[index]->next;
+		index++;
+	}
+
+	if (scene_items_cached[items_count-1] == NULL || scene_items_cached[items_count-1]->next != NULL) {
```

Comment:
Since `scene_items_cached[items_count-1]` represents the last item, isn't `scene_items_cached[items_count-1]->next` supposed to be null?

Reply:
Yes. And if it is not a null it means function was called with incorrect number of items.

Reply:
Got it, sounds good